enhancements/certificate-short-rotation.md

```diff
@@ -0,0 +1,132 @@
+---
+title: certificate-short-rotation
+authors:
+  - vrutkovs
+reviewers:
+  - deads2k
+approvers:
+  - deads2k
+api-approvers:
+  - deads2k
+creation-date: 2024-08-24
+last-updated: 2024-08-24
+tracking-link:
+  - https://issues.redhat.com/browse/API-1688
+---
+
+# Short Rotation Period For Certificates
+
+## Summary
+
+Add new feature gate in DevPreview set so that components would issue certificates with shorter 
+duration - hours instead of days.
+
+## Motivation
+
+Currently certificates are issued by Openshift with various validity durations, but at least its 15 
+days. This makes testing certificate rotation in CI complicated - we have to emulate passing time 
+using time skewing. This methods shows how cluster recovers after certificates have expired, but 
+it doesn't help us with testing happy path when certificates rotate during standard cluster lifecycle.
+
+Some components (i.e. cluster-kube-apiserver-operator) issue certificate with shorter lifetime in 
+development branch. This requires us to revert this change every time we branch for new release.
+This also doesn't help us in CI, as it needs a similar change in the installer. 
+Also, most components are not using this, so we end up with some certificates valid for hours but  
+most would be valid for days.
+
+Since the change to revert this setting requires manual pull request, there is chance that this 
+setting will leak into supported releases.
+
+This enhancement describes a new feature gate, which would enable this feature for all components 
+and ensure that stable releases don't have it accidentally enabled as it uses FeatureGates.
+
+### User Stories
+
+> As an Openshift developer, I want to have a setting for component to issue shorter living 
+> certificates so that I could verify that certificate rotation doesn't cause issues
+
+### Goals
+
+* Create a new FeatureGate in DevPreview featureset
+* Update components owning certificates to check this featuregate and issue shorter certificates
```

Comment:
can we add some details as to how an operator would check this featuregateand issue a shorter cert, details like:
- when the operator loads, or part of a control loop
- how do we communicate the duration
- scope, how many such operators we have to change?

Reply:
>when the operator loads, or part of a control loop

Probably part of the control loop - as featuregates can be enabled at any time. Not sure if this is useful anyway, as the component would have to wait for existing certificates to expire (which is minimum of 15 days)

> how do we communicate the duration

Components would hardcode short duration for each signer/target cert. There is no need to make it configurable (as its not useful for CI), just "long/short duration" setting should be sufficient.

> scope, how many such operators we have to change?

So far I counted 8:
* https://github.com/openshift/installer/pull/8784
* https://github.com/openshift/service-ca-operator/pull/245
* https://github.com/openshift/cluster-kube-apiserver-operator/pull/1718
* https://github.com/openshift/cluster-kube-controller-manager-operator/pull/817
* https://github.com/openshift/cluster-network-operator/pull/2470
* https://github.com/openshift/cluster-etcd-operator/pull/1315
* OLM
* possibly Ingress (not sure if necessary, router target cert is not being refreshed)

Briefly tested this and cluster with these PRs applied managed to refresh all certs - however I didn't run conformance tests during/after refreshes.